### PR TITLE
add v100-docs

### PR DESCRIPTION
**Pull request by ypm1999**
* [ x ] Tested
* [ x ] Formatted with YAPF

### CODE REVIEW

File: docs/mlsys21/v100_throughputs.md [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+
+### Amazon EC2
+
+If you have never used Amazon EC2 before, you should request an account from <https://aws.amazon.com/ec2> to get access to the GPU VMs
+
+After getting access, navigate to the "EC2 Dashboard" -> "Launch instance"  pane to create an VM with V100 GPUs.
+- The GPU instance we used for accessing V100 GPUs on Amazon EC2 is [`p3.2xlarge`](https://aws.amazon.com/ec2/instance-types/p3/).
+- The instance is configured as a complete machine with 1 NVIDIA V100 GPU. If you want use more, you can select other larger `p3` instance.
+- We used `NVIDIA Deep Learning AMI v20.06.3`
+- The host resource contains 8 vCPUs, 61 GB memory, but docker instances will limit this.
```
Comment (wangshangsam) [inline docs/mlsys21/v100_throughputs.md:29]:
```
The p3.2xlarge instance contains 8 vCPUs and 61 GB host memory. 
If you selected a larger instance with more GPUs, docker can limit the amount 
of host resource allocated per GPU.
```

File: docs/mlsys21/v100_throughputs.md [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+# clone the code base
+git clone https://github.com/UofT-EcoSystem/hfta.git
+cd hfta
+```
+
+### Acquiring NVIDIA Nsight Compute CLI and DCGM
+
+We require two installation files (`.deb`) for Nsight Compute and DCGM pre-downloaded to build the docker image.
+
+- Nsight Compute: version `3.1.72`, downloaded under `third_party/nsys/nsys_cli_2020.3.1.72.deb`
```
Comment (wangshangsam) [inline docs/mlsys21/v100_throughputs.md:53]: This is actually Nsight System, not Nsight Compute.

File: docs/mlsys21/v100_throughputs.md [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+- DCGM version: version `2.0.10`, downloaded under  `third_party/dcgm/datacenter-gpu-manager_2.0.10_amd64.deb`
+
+In order to download the `.deb` files, you need to register a NVIDIA developer account via: <https://developer.nvidia.com/login>, after that, you can download the .deb file:
+- NVIDIA Nsight Computer at: <https://developer.nvidia.com/nsight-compute>
```
Comment (wangshangsam) [inline docs/mlsys21/v100_throughputs.md:57]: Should be Nsight System.

File: docs/mlsys21/v100_throughputs.md [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+- NVIDIA DCGM at: <https://developer.nvidia.com/dcgm>
+
+
+### Download and launch docker image
+
+Follow the commands below to prepare and launch the docker image, this will take approximately 10 mins
+
+#### Build docker image
+```bash
+# build the image, select from native1.6-cu10.2, nvidia20.06, nvidia20.08
```
Comment (wangshangsam) [inline docs/mlsys21/v100_throughputs.md:67]: V100 and RTX6000 should only select `native1.6-cu10.2`

File: docs/mlsys21/v100_throughputs.md [hunk 5]
```diff
@@ -0,0 +1,168 @@
...
+# this will take about 10 mins to complete
+bash docker/build.sh <the version of the image, e.g. native1.6-cu10.2>
+```
```
Comment (wangshangsam) [inline docs/mlsys21/v100_throughputs.md:70]:
```
#### Reuse prebuilt docker image
If you do not wish to build the docker image from scratch, you can reuse the
prebuilt docker image that we provide:
` ` ` 
docker pull wangshangsam/hfta:mlsys21_native1.6-cu10.2
docker tag wangshangsam/hfta:mlsys21_native1.6-cu10.2 hfta:dev
` ` ` 
```

File: docs/mlsys21/v100_throughputs.md [hunk 6]
```diff
@@ -0,0 +1,168 @@
...
+
+#### Launch a docker container
+```bash
+# launch the image
+# you will need to provide a placeholder mount point for data directory
```
Comment (wangshangsam) [inline docs/mlsys21/v100_throughputs.md:75]: `for data directory` -> `for the data directory`

File: docs/mlsys21/v100_throughputs.md [hunk 7]
```diff
@@ -0,0 +1,168 @@
...
+benchmarks/workflow.sh" expects 3 arguments in total
+- DEVICE: {cuda, tpu}, default is `cuda`
+- DEVICE_MODEL: {a100, v100, etc, or v2/v3 for tpu}, default is `v100`
+- OUTDIR_ROOT: {the output directory of the benchmarked results}, default is `benchmarks`
+
+The first two arguments, are often needed in every run,. Please refer to the script for the usage of the arguments
+```bash
+# The command below will set the target device and device model to be CUDA, A100,
+# and the output directory by default is ./benchmarks, but you can specify other things
+source benchmarks/workflow.sh cuda a100 <optional: output root dir of the benchmark output>
```
Comment (wangshangsam) [inline docs/mlsys21/v100_throughputs.md:148]: This is a tutorial for V100 or RTX6000, so the example `source` command should be 
```
source benchmarks/workflow.sh cuda v100
source benchmarks/workflow.sh cuda rtx6000
```

File: docs/mlsys21/v100_throughputs.md [hunk 8]
```diff
@@ -0,0 +1,168 @@
...
+
+After the workflow experiment is done, run bash function below to process the output and plot the speedup curves. The plot functions are also defined in `_workflow_<modelname>.sh` files.
+
+```bash
+# In general, plot functions are defined as plot_<exp name>
+# For example, for BERT experiment, run
+plot_bert
+```
+
+Finally, you should be able to see the `.csv` and `.png` files under the output directory (`./benchmarks (or the directory you specified above)`).
```
Comment (wangshangsam) [inline docs/mlsys21/v100_throughputs.md:168]: What about the partially fused resnet experiment?

**Review by wangshangsam — CHANGES_REQUESTED** [review]

File: docs/mlsys21/v100_throughputs.md [hunk 9]
```diff
@@ -0,0 +1,168 @@
...
+
+After getting access, navigate to the "EC2 Dashboard" -> "Launch instance"  pane to create an VM with V100 GPUs.
+- The GPU instance we used for accessing V100 GPUs on Amazon EC2 is [`p3.2xlarge`](https://aws.amazon.com/ec2/instance-types/p3/).
+- The instance is configured as a complete machine with 1 NVIDIA V100 GPU. If you want use more, you can select other larger `p3` instance.
+- We used `NVIDIA Deep Learning AMI v20.06.3`
+- The host resource contains 8 vCPUs, 61 GB memory, but docker instances will limit this.
+- We used a standard EBS boot disk with 100GB of storage capacity.
+
+__Note: please make sure to turn off your VM instances as soon as you finish the experiments, as the instances are quite costly__
+
```
[the remark this answers is not in the document]
Comment (nixli) [reply docs/mlsys21/v100_throughputs.md:33]: Driver Version: 450.66 , driver CUDA version 11.0
Description:	Ubuntu 18.04.5 LTS
Docker version 20.10.2, build 2291f61
nvidia-container-runtime version  runc version 1.0.0-rc92, commit: ff819c7e9184c13b7c2607fe6c30ae19403a7aff, spec: 1.0.2-dev
GPU: Quadro RTX 6000 24GB

File: docs/mlsys21/v100_rtx6000_throughputs.md.md [hunk 10]
```diff
@@ -0,0 +1,193 @@
...
+
+
+#### Launch a docker container
+```bash
+# launch the image
+# you will need to provide a placeholder mount point for the data directory
+# default is under ${HOME}/datasets
+ubuntu@ip-xxxxxxxx:~/hfta$ bash docker/launch.sh <optional: data directory mount point> <optional: image tag>
+
+=============
```
Comment (wangshangsam) [inline docs/mlsys21/v100_rtx6000_throughputs.md.md:94]: The output below is actually not what you would see from an `native1.6-cu10.2` image (notice the `Various files include modifications (c) NVIDIA CORPORATION. All rights reserved.` here, which means NVIDIA-built NGC images). 

Just remove the output.

**Review by wangshangsam — CHANGES_REQUESTED** [review]

File: docs/mlsys21/v100_throughputs.md (hunk 9 above)
[the remark this answers is not in the document]
Comment (wangshangsam) [reply docs/mlsys21/v100_throughputs.md]: Hmmm... what exactly is `runc version 1.0.0-rc92, commit: ff819c7e9184c13b7c2607fe6c30ae19403a7aff, spec: 1.0.2-dev`? As in is it `1.0.0-rc92` or `1.0.2-dev`? I don't quite understand this part.

[the remark this answers is not in the document]
Comment (nixli) [reply docs/mlsys21/v100_throughputs.md]: sorry for the confusion, the correct version should be:
2.5.0-1
As shown by 
```bash
dpkg -l | grep nvidia-docker2
```

**Review by wangshangsam — APPROVED** [review]
LGTM!

**Pull request merged**